### ISSUE TITLE
Docker instructions don't work

### ISSUE THREAD

**Issue by CSEdd** [input-output-hk/cardano-db-sync#655]
If you follow the docker instructions (https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/docker.md) on a fresh clone of `10.0.0` you get the following error: 

```
Connecting to network: mainnet
/nix/store/6f4mj582ymmn8zx90j8f8n9sx540hi9x-entry-point/bin/entry-point: line 26: /nix/store/74ky2ipmmh6y7qjxp0q03vh316dxddw2-cardano-db-sync-mainnet: Is a directory
/nix/store/6f4mj582ymmn8zx90j8f8n9sx540hi9x-entry-point/bin/entry-point: line 26: exec: /nix/store/74ky2ipmmh6y7qjxp0q03vh316dxddw2-cardano-db-sync-mainnet: cannot execute: Is a directory
/nix/store/6f4mj582ymmn8zx90j8f8n9sx540hi9x-entry-point/bin/entry-point: line 26: /nix/store/74ky2ipmmh6y7qjxp0q03vh316dxddw2-cardano-db-sync-mainnet: Is a directory
/nix/store/6f4mj582ymmn8zx90j8f8n9sx540hi9x-entry-point/bin/entry-point: line 26: exec: /nix/store/74ky2ipmmh6y7qjxp0q03vh316dxddw2-cardano-db-sync-mainnet: cannot execute: Is a directory
```

To be more specific, I used the following commands:

```
git clone https://github.com/input-output-hk/cardano-db-sync
cd cardano-db-sync
# optional - git checkout tags/10.0.0
docker-compose up -d && docker-compose logs -f
```

You'll see db-sync fail to get started with the above error until it eventually stops.

**Comment by erikd** [comment]
@rhyslbw  ?

**Comment by CSEdd** [comment]
> @rhyslbw ?

This might be more of a documentation issue actually - it doesn't work on `tags/9.0.0` either but does_on `<= tags/8.0.0`

**Comment by erikd** [comment]
My understanding is that this in an interaction between Nix and Docker.

**Comment by meiblorn** [comment]
I confirm, docker `entry-point` is broken since 10.0.0 version.

**9.0.0**
Run at host:
```
docker run -it --rm --entrypoint /bin/bash inputoutput/cardano-db-sync:9.0.0
```
Then run inside docker container:
```
bash-4.4# cat /bin/entry-point
.... select some path to executable ......

bash-4.4# /nix/store/03sjsjjdlgby9zb1ac1yb3v5800h8qzg-cardano-db-sync
cardano-db-sync-extended: Environment variable 'PGPASSFILE not set.
CallStack (from HasCallStack):
  error, called at src/Cardano/Db/PGConfig.hs:56:16 in cardano-db-9.0.0-T4gnjhGtZS6X2sIpr0asH:Cardano.Db.PGConfig
```
It executes normally

**10.0.0**
```
docker run -it --rm --entrypoint /bin/bash inputoutput/cardano-db-sync:10.0.0
```
```
bash-4.4# cat /bin/entry-point
.... select some path to executable ......

bash-4.4# /nix/store/a2d25qa2b8x7mynnwmmdxkbij78b9m7w-cardano-db-sync-testnet
bash: /nix/store/a2d25qa2b8x7mynnwmmdxkbij78b9m7w-cardano-db-sync-testnet: Is a directory

bash-4.4# ls /nix/store/a2d25qa2b8x7mynnwmmdxkbij78b9m7w-cardano-db-sync-testnet/bin/cardano-db-sync-testnet 
/nix/store/a2d25qa2b8x7mynnwmmdxkbij78b9m7w-cardano-db-sync-testnet/bin/cardano-db-sync-testnet
```
It doesn't work because these paths are directories with `/bin/xyz` directories inside

**Comment by erikd** [comment]
Can this be closed?

**Comment by CSEdd** [comment]
@erikd This is still an issue as replicated by @meiblorn. It looks like Nix changed its paths so previous executables are now directories. Without a fix we need to use an old version of `db-sync` and `node:1.25.1` to get around it via Docker.

**Comment by erikd** [comment]
This is fixed in the `10.0.1` release (and this docker fix is the only change from `10.0.0`).
